### DISCUSSION TITLE
[Bug]: Why on server is not running when i use bullmq?

### DISCUSSION THREAD

**Discussion by xts-bit** [taskforcesh/bullmq#2104] [Q&A] [answered]
As i start my server it logs Working, and Job completed why job running as i start the server, it should wait for 5 seconds and then run that function...

```
const queue = new Queue('myQueue', {
  connection: {
    host: 'host',
    port: 11525,
    password: 'pass'
  }
});

async function hitApi() {
  console.log("Working")
}

queue.add('myJob', { delay: 5000 });

const worker = new Worker('myQueue', async (job) => {
  await hitApi();
}, {
   connection: {
    host: 'host',
    port: 11525,
    password: 'pass'
  }
});

worker.on('completed', (job) => {
  console.log(`Job completed`);
});

worker.on('failed', (job, err) => {
  console.error(`Job failed for URL`);
});
```

**Comment by roggervalf** [comment]
Hi @xts-bit, you may need to read our docs https://docs.bullmq.io/readme-1, you must have a worker instance

**Comment by xts-bit** [comment]
@roggervalf Can you please give a code example to run a function after 10 seconds? even it server is restarted many timeS?

**Reply by manast** [reply]
Here you should find information on how to run delayed jobs: https://docs.bullmq.io/guide/jobs/delayed

**Reply by xts-bit** [reply]
@manast Can you give me a code example here with Worker and Queue?

**Reply by manast** [reply]
There are many examples, and tutorials, in the official documentation and https://blog.taskforce.sh

**Reply by xts-bit** [reply]
@manast I already checked them but I am new and don't understand docs etc...  i faced that issue

**Reply by manast** [reply]
You can start from one of the tutorials and build from there. If you have specific questions we are happy to help.

**Reply by xts-bit** [reply]
@manast i tried this facing this issue in which we are commenting...

**Reply by manast** [reply]
The code you posted does not come from any official documentation or working examples.

**Reply by xts-bit** [reply]
@manast Please check my edited post🙏

**Reply by manast** [reply]
What is the issue?

**Reply by xts-bit** [reply]
@manast As i start my server it logs Working, and Job completed why job running as i start the server, it should wait for 5 seconds and then run that function... 

But the expected behavior should be it runs that function after 5 seconds.

**Reply by manast** [reply, marked as answer]
Ok, you are not calling ```Queue.add``` with its proper arguments. Please use Typescript or some other tool to help you avoid this kind of issues in the future:
```ts
queue.add("myJob", { foo, "bar" }, { delay: 5000 });
```

**Reply by xts-bit** [reply]
@manast Yes this works, But will my code will work when the server is restarted? like as i push new code to the server it restarts the server so will this cause any problem? in my case, the delay is around 1 day.

**Reply by xts-bit** [reply]
@manast  please check this https://github.com/taskforcesh/bullmq/discussions/2114
